tests/suites/test_suite_pk.function

```diff
@@ -715,6 +728,10 @@ void pk_ec_test_vec(int type, int id, data_t *key, data_t *hash,
     TEST_ASSERT(mbedtls_ecp_point_read_binary(&eckey->grp, &eckey->Q,
                                               key->x, key->len) == 0);
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    TEST_ASSERT(mbedtls_pk_update_public_key_from_keypair(&pk) == 0);
```

Comment:
This (and the next 3 similar additions) is a compatibility smell. What our test code is doing here would be perfectly legal (according to our documentation) for an application to do. Applications that were doing that would then be broken after this PR, which is not acceptable.

This PR is already fairly large as it is, so I've created a follow-up issue which I suggest you address right after this PR and #7438 have been merged: #7570

Reply:
Based on the discussion in https://github.com/Mbed-TLS/mbedtls/issues/7570, I think this is not just a compatibility smell, but an outright design blocker. As soon as `mbedtls_pk_ec` is called on the PK context — and [we can't detect that](https://github.com/Mbed-TLS/mbedtls/issues/7570#issuecomment-1542186552) — the `priv_id` field becomes unusable. So the `priv_id` field is effectively unusable unless the `mbedtls_pk_ec` function doesn't exist. Since this function has to exist in all the cases where it exists in Mbed TLS 3.4.0 for backward compatibility, the thing we can do is:

* Make `mbedtls_pk_ec` not part of the public API if `MBEDTLS_ECP_LIGHT` is defined but not `MBEDTLS_ECP_C`. It can still be defined in an internal header, possibly returning a const pointer because we want it to be read-only access.
* Change `priv_id` and all the code that accesses it to be invoked only when `!defined(MBEDTLS_ECP_C)`.

Reply:
>  possibly returning a const pointer because we want it to be read-only access

Some tests use `mbedtls_pk_ec` to get access to the `ecp_keypair` directly to write it, just like the one that @mpg highlighted in this comment. I think that making `mbedtls_pk_ec` private is already enough (we have control on what the library's code does, but not the final user's one) since it allows to keep this test. Wdyt?

Reply:
See https://github.com/Mbed-TLS/mbedtls/issues/7570#issuecomment-1545443804 - I'm trying to regroup all discussion in one place as I find it harder to follow otherwise.